homeassistant/components/rainmachine/__init__.py

```diff
@@ -320,6 +350,19 @@ async def async_pause_watering(call: ServiceCall, controller: Controller) -> Non
         """Pause watering for a set number of seconds."""
         await controller.watering.pause_all(call.data[CONF_SECONDS])
 
+    @call_with_controller(update_programs_and_zones=False)
+    async def async_push_flow_meter_data(
+        call: ServiceCall, controller: Controller
+    ) -> None:
+        """Push flow meter data to the device."""
+        payload = {CONF_VALUE: call.data.get(CONF_VALUE)}
+        units = call.data.get(CONF_UNIT_OF_MEASUREMENT)
+        if units is not None:
+            payload[CONF_UNITS] = units
+        return_code = await controller.watering.post_flowmeter(payload)
```

Comment:
Reminder that this new call doesn't return an `int` – it returns the entire payload.

All that said, you don't need to do error code checking here: `@call_with_controller` will catch an errors and do the correct thing.